term_1/solver.cpp

```diff
@@ -133,8 +133,7 @@ struct KahanVector // Вектор переопределяющий сумму 
 private:
     void collapse()
     {
-        // segmentation fault if value.size() == 0
-        for (auto v = value.begin() + 1; v != value.end(); v++)
+        for (auto v = value.begin() + 1; v != value.end(); v++) // v.size() != 0
```

Comment:
validation of v..size is required in case the code crashes in this case

Reply:
Thanks. Well, v.size() == 0 is logical error and each constructor creates non-empty vector. I can throw logic_error exception in this point, but i think that several validations in every step of calculations will negatively effect the efficiency.

Reply:
Sure. Validations should be done either upon or during the configuring step. I didn't notice that the vectors there was non-empty and was guided mainly by your comments. Adding any run-time checks in core simulator code is definitely not a wise idea in any case. Even for the sake of debugging the asserts would be sufficient (and would not cause any additional operations in release builds).